### PR TITLE
Add transformation api client

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-753

### Change description ###
Added transformation api client and the tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ) {
+        this.restTemplate = restTemplate;
+        this.objectMapper = objectMapper;
+    }
+
+    public SuccessfulTransformationResponse transformExceptionRecord(
+        String baseUrl,
+        ExceptionRecord exceptionRecord,
+        String s2sToken
+    ) throws Exception {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java]: could be ` throws CaseTransformationException` I think

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/TransformationClientTest.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            )
+        );
+    }
+
+    private JSONObject successResponse() throws JSONException {
+        return new JSONObject()
+            .put("case_creation_details", new JSONObject()
+                .put("case_type_id", "some_case_type")
+                .put("event_id", "createCase")
+                .put("case_data", new JSONObject().put("case_field", "soem value")))
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/TransformationClientTest.java]: typo in 'some'

Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/TransformationClientTest.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java]: Yes, almost. Also accompanied with `IOException`

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.UNPROCESSABLE_ENTITY;
+
+@IntegrationTest
+public class TransformationClientTest {
```
[the remark this answers is not in the document]
Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClientTest.java]: test...

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    public TransformationClient(
+        RestTemplate restTemplate,
+        ObjectMapper objectMapper
+    ) {
+        this.restTemplate = restTemplate;
+        this.objectMapper = objectMapper;
+    }
+
+    public SuccessfulTransformationResponse transformExceptionRecord(
+        String baseUrl, // todo base url as part of constructor grabbed from config
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java]: should the 'todo' stay?
There's a different url for every service, do we want to create multiple clients??

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/transformation/TransformationClient.java]: oh yes. i'll remove the todo and if we see it is better to have a map or so in future - we can revisit

**Pull request merged**